File: Easy/0242_valid-anagram/valid-anagram.py
#!/bin/python3

from typing import List
from collections import Counter
from dataclasses import dataclass

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)

@dataclass
class TestCase:
    s: str
    t: str

def main():
    tests: List[TestCase] = [
        TestCase("anagram", "nagaram"),
        TestCase("rat", "car"),
    ]

    for tc in tests:
        print(f"---\ns: '{tc.s}', t: '{tc.t}'")
        print(f"isAnagram: {Solution().isAnagram(tc.s, tc.t)}")

if __name__ == "__main__":
    main()

File: Easy/0242_valid-anagram/valid-anagram.go
package main

import (
	"fmt"
)

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	var count [26]int
	for _, ch := range s {
		count[ch - 'a']++
	}
	for _, ch := range t {
		count[ch - 'a']--
		if count[ch - 'a'] < 0 {
			return false
		}
	}
	return true
}

func main() {
	tests := [][]string{
		{"anagram", "nagaram"},
		{"rat", "car"},
	}

	for _, tc := range tests {
		fmt.Printf("---\ns: %q, t: %q\nisAnagram: %t\n", tc[0], tc[1], isAnagram(tc[0], tc[1]))
	}
}
